### DISCUSSION TITLE
wlr_scene_graph API usage

### DISCUSSION THREAD

**Discussion by Shinyzenith** [qtile/qtile#3721] [Ideas] [answered]
The following api can simplify a lot of the codebase and remove damage tracking related clutter as it gives us damage tracking for free :D
It's basically a catch-all rendering api. 

I would love to work on a PR if this is in scope for the project. I am aware that some compositors opt-out of it due to the fact that you cannot manipulate egl-textures with wlr_scene_graph ( If I'm not mistaken )

**Comment by m-col** [comment, marked as answer]
This has been discussed a few times (might be worth a search, though some discussion may have been in the chatroom). Tl;dr: it's been on the roadmap for a while and a couple of us are very eager for this (I remember @jwijenbergh and I have discussed it and agree it's a great idea). However nobody yet has started working on it, so if you're interesting in working on this then that's fantastic and thank you for that. I suspect it will fix a number of bugs, such as https://github.com/qtile/qtile/issues/2544 and a few visual glitches that are related to that.

**Reply by Shinyzenith** [reply]
> This has been discussed a few times (might be worth a search, though some discussion may have been in the chatroom). Tl;dr: it's been on the roadmap for a while and a couple of us are very eager for this (I remember @jwijenbergh and I have discussed it and agree it's a great idea). However nobody yet has started working on it, so if you're interesting in working on this then that's fantastic and thank you for that. I suspect it will fix a number of bugs, such as #2544 and a few visual glitches that are related to that.

Awesome!
I just wanna preface the PR with the fact that I might not be able to finish it all alone in a short time-span due to the size of this herculean task. So if anyone would like to chime in ( Maybe you? :D) that would really help to get the ball rolling and eventually merge the patch in master branch.

**Reply by m-col** [reply]
No worries at all, and while I'm very eager to have this API in the works, I should add my own preface that until around Oct I will be extremely limited in how much time and work I can contribute to this (and Qtile in general...). So really no rush or deadline or anything like that! But of course in principle I'm happy to help out, though my current understanding of the wlr_scene_graph API is pretty surface-level... I'll need to update myself.

**Reply by Shinyzenith** [reply]
Cool! I'll sketch up a general layout of my plan and start the PR soon 😄

**Reply by m-col** [reply]
Awesome :)

**Reply by jwijenbergh** [reply]
I have been wanting to implement this for a while but haven't had the time really. I am happy to assist with the time I have, so if you have any questions let me know (note: I am also not an expert here, but have some general knowledge about the api)
